packages/react-refresh/src/__tests__/ReactFresh-test.js

```diff
@@ -2448,16 +2464,15 @@ describe('ReactFresh', () => {
       };
     });
 
-    const root = ReactDOMClient.createRoot(container);
     root.render(<AppV1 offscreen={true} />);
     await waitFor(['App#layout']);
     const el = container.firstChild;
     expect(el.hidden).toBe(true);
     expect(el.firstChild).toBe(null); // Offscreen content not flushed yet.
 
     // Perform a hot update.
-    patch(() => {
-      function Hello() {
+    {
+      const Hello = () => {
```

Comment:
When I change this to `await patch` i.e. wrapping `performReactRefresh` in `act` I get considerably different assertions. Seems like this test wants to assert on some intermediate states but I don't know how to port them